pkg/cmd/pulumi/import.go

```diff
@@ -0,0 +1,549 @@
+		}
+
+		if _, err := w.Write(contents); err != nil {
+			return err
+		}
+		return nil
+	}, resources, names)
+}
+
+func newImportCmd() *cobra.Command {
+	var parentSpec string
+	var providerSpec string
+	var importFilePath string
+	var outputFilePath string
+
+	var debug bool
+	var message string
+	var stack string
+	var execKind string
+
+	// Flags for engine.UpdateOptions.
+	var diffDisplay bool
+	var eventLogPath string
+	var parallel int
+	var showConfig bool
+	var skipPreview bool
+	var suppressOutputs bool
+	var suppressPermaLink bool
+	var yes bool
+	var protectResources bool
+
+	cmd := &cobra.Command{
+		Use:   "import [type] [name] [id]",
+		Args:  cmdutil.MaximumNArgs(3),
+		Short: "Import resources into an existing stack",
+		Long: "Import resources into an existing stack.\n" +
+			"\n" +
+			"Resources that are not managed by Pulumi can be imported into a Pulumi stack\n" +
+			"using this command. A definition for each resource will be printed to stdout\n" +
+			"in the language used by the project associated with the stack; these definitions\n" +
+			"should be added to the Pulumi program. The resources are protected from deletion\n" +
+			"by default.\n" +
+			"\n" +
+			"Should you want to import your resource(s) without protection, then you can pass\n" +
+			"`--protect=false` as an argument to the command. This will mark all resources as unprotected" +
+			"\n" +
+			"\n" +
+			"A single resource may be specified in the command line arguments or a set of\n" +
+			"resources may be specified by a JSON file. This file must contain an object\n" +
+			"of the following form:\n" +
+			"\n" +
+			"    {\n" +
+			"        \"nameTable\": {\n" +
+			"            \"provider-or-parent-name-0\": \"provider-or-parent-urn-0\",\n" +
+			"            ...\n" +
+			"            \"provider-or-parent-name-n\": \"provider-or-parent-urn-n\",\n" +
+			"        },\n" +
+			"        \"resources\": [\n" +
+			"            {\n" +
+			"                \"type\": \"type-token\",\n" +
+			"                \"name\": \"name\",\n" +
+			"                \"id\": \"resource-id\",\n" +
+			"                \"parent\": \"optional-parent-name\",\n" +
+			"                \"provider\": \"optional-provider-name\",\n" +
+			"                \"version\": \"optional-provider-version\",\n" +
+			"            },\n" +
+			"            ...\n" +
+			"            {\n" +
+			"                ...\n" +
+			"            }\n" +
+			"        ]\n" +
+			"    }\n" +
+			"\n" +
+			"The name table maps language names to parent and provider URNs. These names are\n" +
+			"used in the genrated definitions, and should match the corresponding declarations\n" +
+			"in the source program. This table is required if any parents or providers are\n" +
+			"specified by the resources to import.\n" +
+			"\n" +
+			"The resources list contains the set of resources to import. Each resource is\n" +
+			"specified as a triple of its type, name, and ID. The format of the ID is specific\n" +
```

Comment:
I think we can be a bit clearer about what type and name are. Otherwise, it might be confusing what is meant by 'type' here. Similarly, we should be clear how 'name' is used in the resulting generated program.